src/state/download/actions.js

```diff
@@ -174,58 +175,34 @@ export const addExperimentSucceeded = (dataSetId, dataSet) => {
 export const fetchDataSet = () => {
   return async dispatch => {
     const dataSetId = localStorage.getItem('dataSetId');
+    if (!dataSetId) {
+      return dispatch(fetchDataSetSucceeded({}));
```

Comment:
Why are we calling this success function when there's no `dataSetId` found?

Reply:
hmm yes, I refactored this code to remove one condition, before it was:

```
    const dataSetId = localStorage.getItem('dataSetId');
    [...]
    const response = dataSetId
      ? await asyncFetch(`/dataset/${dataSetId}/`)
      : null;

    const dataSet = response ? response.data : {};

    dispatch(fetchDataSetSucceeded(dataSet));
```

So even when `dataSetId` is not defined it still calls the success function with an empty object. Should we remove this part??

Reply:
haha my bad. Yes, my gut instinct is that we should only call the success function when there is a dataSetId? What do you think?